obsplus/datasets/dataset.py

```diff
@@ -175,13 +193,26 @@ def _load(self, what, path):
     def copy(self, deep=True):
```

Comment:
Maybe include the type?
`def copy(self, deep: bool = True):`

Reply:
ya, it looks like there are a few inconsistencies/missing type hints in this module.